### ISSUE TITLE
Worker abnormal exit

### ISSUE THREAD

**Issue by pantaovay** [swoole/swoole-src#342]
## Desc
I am using swoole and websocket. If I call the `close` function in the `onOpen` callback function, the worker will exit abnormally. The error message is:
```
WARN    swFactoryProcess_manager_loop: worker#3 abnormal exit, status=0, signal=11
```

The debug log:
```
[/data/install/swoole-src-swoole-1.7.14-stable/src/factory/FactoryProcess.c:729@swFactoryProcess_finish][Worker] send: sendn=201|type=0|content=HTTP/1.1 101 Switching Protocols
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Accept: 6wEyfpmvfZN4c0bmIfWf1oijjsk=
Sec-WebSocket-Version: 13
Server: swoole-websocket-server


[/data/install/swoole-src-swoole-1.7.14-stable/swoole_websocket.c:83@swoole_websocket_onOpen]



conn ws status:2, fd=1



[/data/install/swoole-src-swoole-1.7.14-stable/swoole_websocket.c:87@swoole_websocket_onOpen]



handshake success



[/data/install/swoole-src-swoole-1.7.14-stable/src/factory/FactoryProcess.c:729@swFactoryProcess_finish][Worker] send: sendn=12|type=4|content=(null)
[/data/install/swoole-src-swoole-1.7.14-stable/src/network/ReactorThread.c:174@swReactorThread_close]Close Event.fd=38|from=0
[/data/install/swoole-src-swoole-1.7.14-stable/src/network/ReactorThread.c:242@swReactorThread_close]set_maxfd=37|close_fd=38

[2015-08-07 02:18:06 $1941.0]   WARN    swFactoryProcess_manager_loop: worker#3 abnormal exit, status=0, signal=11
```
 
## Replay
And I write a demo that can 100 percent replay the bug:
```php
<?php
$server = new \swoole_websocket_server('192.168.10.2', 9501);
$server->on('open', function ($server, $request) {
    $server->close($request->fd);
});
$server->on('message', function ($server, $frame) {
    // do nothing
});
$server->start();
```
When a client tries to connect the server, swoole will give the above error message.

## Environment

1. Ubuntu 12.04 X64
2. PHP 5.6.8
3. Swoole1.7.14

**Comment by matyhtf** [comment]
请使用最新的1.7.19-rc2测试一下。

**Comment by pantaovay** [comment]
Fixed after changing to 1.7.18👍
